warning in test code compile of h5_test/tst_h_files.c

I am seeing this warning in the h5_test directory make check build:
```
gcc -DHAVE_CONFIG_H -I. -I..  -I../include -I../oc2  -I/usr/local/zlib-1.2.11/include -I/usr/local/szip-2.1/include -I/usr/local/hdf5-1.10.1_memchecker/include  -g -Wall -fsanitize=address -fno-omit-frame-pointer -MT tst_h_files.o -MD -MP -MF $depbase.Tpo -c -o tst_h_files.o tst_h_files.c &&\
mv -f $depbase.Tpo $depbase.Po
tst_h_files.c: In function ‘main’:
tst_h_files.c:35:14: warning: unused variable ‘type_size’ [-Wunused-variable]
       size_t type_size;
              ^~~~~~~~~
```

Fix is easy. PR will be up shortly.

--- a/h5_test/tst_h_files.c
+++ b/h5_test/tst_h_files.c
@@ -32,7 +32,6 @@ main()
       int obj_class;
       char obj_name[STR_LEN + 1];
       H5T_class_t class;
-      size_t type_size;
       int j, k;
       hid_t tmp1;
 

